R/weight-streetnet.R

```diff
@@ -1,9 +1,64 @@
 #' Weight a street network according to a specified weighting profile.
 #'
-#' Weight (or re-weight) an \pkg{sf} or `SC` (`silicate`)-formatted OSM street
-#' network according to a named profile, selected from (foot, horse, wheelchair,
-#' bicycle, moped, motorcycle, motorcar, goods, hgv, psv), or a customized
-#' version derived from those.
+#' Weight (or re-weight) an \pkg{sf} or \pkg{silicate} *("SC") formatted OSM
+#' street network according to a named profile, selected from (foot, horse,
+#' wheelchair, bicycle, moped, motorcycle, motorcar, goods, hgv, psv), or a
+#' cusstomized version dervied from those.
+#'
+#' @details The structure of networks generated by this function is dependent
+#' on many aspects of the input network, and in particular on specific
+#' key-value pairs defined in the underlying OpenStreetMap (OSM) data.
+#'
+#' OSM keys which can be used to determine restrictions, and so may be added to
+#' the `keep_cols` argument, include:
+#' \itemize{
+#' \item "access"
+#' \item "bicycle"
+#' \item "foot"
+#' \item "highway"
+#' \item "motorcar"
+#' \item "motor_vehicle"
+#' \item "oneway"
+#' \item "restriction"
+#' \item "toll"
+#' \item "vehicle"
+#' }
+#'
+#' Realistic routing including factors such as access restrictions, turn
+#' penalties, and effects of incline, can only be implemented when the objects
+#' passed to `weight_streetnet` are of \pkg{sc} ("silicate") format, generated
+#' with \link{dodgr_streetnet_sc}. Restrictions applied to ways (in OSMM
+#' terminology) may be controlled by ensuring specific columns are retained in
+#' the `dodgr` network with the `keep_cols` argument. For example, restrictions
```

Comment:
This behaviour holds only with `sc` objects, right? If so, I would document which fields influence only `sf` objects (I guess none), which fields influence only `sc` objects, which fields influence both.

Reply:
The whole list of keys has now been completely rewritten again. The biggest difference between sf and sc is in turn restrictions and effects of elevation, both of which are only possible in sc. This is now explicitly explained. Both of those work without specifying any additional `keep_cols`. I'll resolve this comment now, and ask for your review on this PR again, thanks